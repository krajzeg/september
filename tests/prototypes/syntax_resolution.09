t := Object()
t:::"[]" = |index| { 42 }
t:::test1 = { [1] }
t:::test2 = { this[1] }

print("First test should create an array, so this should be true:", t.test1().is(Array))
print("Second test should return the result of indexing, so this should be 42:", t.test2())

