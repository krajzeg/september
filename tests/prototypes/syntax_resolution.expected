First test should create an array, so this should be true: <True>
Second test should return the result of indexing, so this should be 42: 42
